File: typescript/tracing/openrouter/01_chat_completion.ts
import { CHAT_MODEL, createOpenRouterClient, createRespan, logExampleResult, runWithOpenRouterWorkflow, shutdownRespan } from "./_shared.js";

type OpenRouterRespan = ReturnType<typeof createRespan>;

export async function runChatCompletion(respan: OpenRouterRespan = createRespan()): Promise<string | undefined> {
  const workflowName = "openrouter_ts_chat_completion";
  const shouldShutdown = arguments.length === 0;
  try {
    const content = await runWithOpenRouterWorkflow(respan, workflowName, async () => {
      const client = createOpenRouterClient();
      const response = await client.chat.send({
        chatRequest: {
          model: CHAT_MODEL,
          messages: [
            { role: "system", content: "You answer with short, concrete responses." },
            { role: "user", content: "Say hello from OpenRouter and Respan in one sentence." },
          ],
          temperature: 0,
          maxTokens: 80,
        },
      });
      return response.choices?.[0]?.message?.content;
    });
    logExampleResult(workflowName, { model: CHAT_MODEL, content });
    return content;
  } finally {
    if (shouldShutdown) await shutdownRespan(respan);
  }
}

if (import.meta.url === `file://${process.argv[1]}`) {
  runChatCompletion().catch((error) => {
    console.error(error);
    process.exitCode = 1;
  });
}

File: python/tracing/openrouter/01_chat_completion.py
"""OpenRouter chat completion through the OpenAI-compatible Python client."""

from _shared import close_sync, make_client, make_respan
from respan import workflow


def main() -> None:
    respan = None
    client = None
    try:
        respan = make_respan(scenario="basic_chat")
        client, model = make_client()

        @workflow(name="openrouter_basic_chat")
        def run(prompt: str) -> str:
            response = client.chat.completions.create(
                model=model,
                messages=[{"role": "user", "content": prompt}],
            )
            return response.choices[0].message.content or ""

        print(run("Say hello from OpenRouter in one concise sentence."))
    finally:
        close_sync(respan=respan, client=client)


if __name__ == "__main__":
    main()
